### PR TITLE
Update docker-compose.yaml

### PR DESCRIPTION
**Pull request by trondhindenes**
update docker-compose.yaml to it matches the one in https://github.com/trondhindenes/openfga.dev/blob/main/docs/content/getting-started/setup-openfga.mdx

## Description
Makes docker-compose.yaml match the one in the getting started guide (https://github.com/trondhindenes/openfga.dev/blob/main/docs/content/getting-started/setup-openfga.mdx)

## References
- https://github.com/openfga/openfga.dev/pull/205

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Review by rhamzeh — APPROVED** [review]

**Review by jon-whit — CHANGES_REQUESTED** [review]
Why do we need the guard on `db.condition.service_started`? The `migrate` guard already covers our bases does it not?

If the migration service has run, then the postgres service is already healthy.

File: docker-compose.yaml [hunk 1]
```diff
@@ -1,40 +1,61 @@
...
-    container_name: postgres
     networks:
-      - default
-    ports:
-      - "5432:5432"
+      - openfga
     environment:
       - POSTGRES_USER=postgres
       - POSTGRES_PASSWORD=password
+      - POSTGRES_DB=openfga
```
Comment (craigpastro) [inline docker-compose.yaml:14]: If this is changed, this postgres container will no longer work with `make run-postgres`. I think it is nice to be able to `docker compose up -d postgres` and then `make run-postgres`.

**Review by craigpastro — CHANGES_REQUESTED** [review]

File: docker-compose.yaml [hunk 2]
```diff
@@ -1,40 +1,61 @@
...
     healthcheck:
       test: [ "CMD-SHELL", "pg_isready -U postgres" ]
       interval: 5s
       timeout: 5s
       retries: 5
 
+  pause:
```
[the remark this answers is not in the document]
Comment (trondhindenes) [reply docker-compose.yaml:21]: In my experience it's not completely robust. `pause` is definetely a hack, but in my experience a needed one.

File: docker-compose.yaml (hunk 1 above)
Comment (trondhindenes) [reply docker-compose.yaml:14]: I'm not sure what that means. Was the default db name of `postgres` used in the past?

**Comment by trondhindenes** [comment]
this PR has been stuck for a bit, I'm not quite sure what to do. It started with me just updating the corresponding compose-file in docs to a working example (https://github.com/openfga/openfga.dev/pull/205) and I was asked in the comment to make the same changes in this repo (which is the current PR). So I did that but it feels like a bit of can of worms. I guess the compose-file in docs and the one in this repo should be in sync, which they currently aren't. I also understand that changing the compose-file in this repo has other implications. So I guess I'm asking:

- should the compose-file in the docs repo and this one be kept in sync at all?
- if so, it's probably better if the docs one replicates this one instead of the other way around

**Pull request closed without merging**